src/client/options/mod.rs

```diff
@@ -523,8 +523,10 @@ pub struct ClientOptions {
     #[builder(default, setter(skip))]
     pub(crate) zlib_compression: Option<i32>,
 
+    /// Information from the SRV URI that generated these client options, if applicable.
     #[builder(default, setter(skip))]
-    pub(crate) original_srv_hostname: Option<String>,
+    #[serde(skip)]
+    pub(crate) original_srv_info: Option<OriginalSrvInfo>,
```

Comment:
originally we were just storing the SRV hostname, but we also need the [min TTL](https://github.com/mongodb/specifications/blob/master/source/polling-srv-records-for-mongos-discovery/polling-srv-records-for-mongos-discovery.rst#should-dns-polling-use-heartbeatfrequencyms-or-dns-cache-ttls) to know how long to sleep for before doing another poll.